lib/metasploit/framework/data_service/proxy/host_data_proxy.rb

```diff
@@ -88,4 +100,25 @@ def valid(opts)
     return true
   end
 
+  def use_id_or_get_id_from_host_address(opts, method)
+    begin
+      if opts[:id]
+        self.data_service_operation do |data_service|
+          data_service.send(method, opts)
+        end
+      elsif opts[:address]
+        self.data_service_operation do |data_service|
+          host = data_service.get_host(opts)
+
+          if host
+            opts[:id] = host.id
+            data_service.send(method, opts.reject {|k,v| k==:address})
+          end
+        end
+      end
+    rescue => e
+      self.log_error(e, "Problem retrieving host")
```

Comment:
Minor: error doesn't seem correct given the functions that will be called

Reply:
Fixed 👍